src/main/java/plugins/nate/smp/Settings.java

```diff
@@ -0,0 +1,157 @@
+package plugins.nate.smp;
+
+import org.bukkit.Material;
+import org.bukkit.configuration.file.FileConfiguration;
+import org.bukkit.configuration.file.YamlConfiguration;
+import org.bukkit.entity.EntityType;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class Settings {
+    private static File configFile;
+    private static FileConfiguration config;
+    public int timberMaxBlocks;
+    public int veinminerMaxBlocks;
+    public List<Material> veinminableBlocks;
+    public List<EntityType> protectedMobs;
+    public int maxAnvilRepairCost;
+    public Set<Material> eligibleContainersForLock;
+    public boolean concretePowderConversion;
+    public boolean cropReplanter;
+
+    public Settings() {
+        configFile = new File(SMP.getPlugin().getDataFolder(), "config.yml");
+        if (!configFile.exists()) {
+            SMP.getPlugin().saveResource("config.yml", false);
+        }
+
+        reloadConfig();
+    }
+
+    public void reloadConfig() {
+        config = YamlConfiguration.loadConfiguration(configFile);
+
+        timberMaxBlocks = _getTimberMaxBlocks();
+        veinminerMaxBlocks = _getVeinminerMaxBlocks();
+        veinminableBlocks = _getVeinminableBlocks();
+        protectedMobs = _getProtectedMobs();
+        maxAnvilRepairCost = _getMaxAnvilRepairCost();
+        eligibleContainersForLock = _getEligibleContainersForLock();
+        concretePowderConversion = _convertConcretePowderInWater();
+        cropReplanter = _autoReplantsCrops();
+    }
+
+    public FileConfiguration getConfig() {
+        return config;
+    }
+
+    public int _getTimberMaxBlocks() {
+        return config.getInt("timber-max-blocks", 192);
+    }
+
+    public int getTimberMaxBlocks() {
+        return timberMaxBlocks;
+    }
+
+    public int _getVeinminerMaxBlocks() {
+        return config.getInt("veinminer-max-blocks", 128);
+    }
+
+    public int getVeinminerMaxBlocks() {
+        return veinminerMaxBlocks;
+    }
+
+    public List<Material> _getVeinminableBlocks() {
+        List<?> rawList = config.getList("veinminable-blocks", new ArrayList<>());
+        List<Material> veinminableBlocks = new ArrayList<>();
+
+        for (Object obj : rawList) {
+            if (obj instanceof String) {
+                try {
+                    Material material = Material.valueOf((String) obj);
+                    veinminableBlocks.add(material);
+                } catch (IllegalArgumentException ignored) {
+
+                }
+            }
+        }
+
+        return veinminableBlocks;
+    }
```

Comment:
```suggestion
    public List<Material> _getVeinminableBlocks() {
        List<String> blockIDs = config.getStringList("veinminable-blocks");
        List<Material> veinminableBlocks = new ArrayList<>();

        for (String blockID : blockIDs) {
            Material material = Material.valueOf(blockID);
            veinminableBlocks.add(material);
        }

        return veinminableBlocks;
    }
```
![image](https://github.com/Coolment-Content-Creation/SMP/assets/22324401/42dc7e1b-f7c5-4efb-b782-9ea22df481d9)